validation-test/Sema/type_checker_perf/slow/mixed_string_array_addition.swift

```diff
@@ -0,0 +1,12 @@
+// RUN: %target-typecheck-verify-swift -swift-version 5 -solver-expression-time-threshold=1
+
+func method(_ arg: String, body: () -> [String]) {}
+
+func test(str: String, properties: [String]) {
+  // expected-error@+1 {{the compiler is unable to type-check this expression in reasonable time}}
```

Comment:
This isn't new, but a case I discovered when debugging the source compat failures. This compiles (very slowly) if you add a type annotation to `param` below.